app/client/cli/helpers/setup.go

```diff
@@ -4,26 +4,37 @@ import (
 	"fmt"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 
 	"github.com/pokt-network/pocket/app/client/cli/flags"
 	"github.com/pokt-network/pocket/logger"
 	"github.com/pokt-network/pocket/p2p"
 	rpcCHP "github.com/pokt-network/pocket/p2p/providers/current_height_provider/rpc"
 	rpcPSP "github.com/pokt-network/pocket/p2p/providers/peerstore_provider/rpc"
 	"github.com/pokt-network/pocket/runtime"
+	"github.com/pokt-network/pocket/runtime/configs"
 	"github.com/pokt-network/pocket/shared/modules"
 )
 
+// TODO_THIS_COMMIT: add godoc comment explaining what this **is** and **is not**
+// intended to be used for.
+const debugPrivKey = "09fc8ee114e678e665d09179acb9a30060f680df44ba06b51434ee47940a8613be19b2b886e743eb1ff7880968d6ce1a46350315e569243e747a227ee8faec3d"
+
 // P2PDependenciesPreRunE initializes peerstore & current height providers, and a
 // p2p module which consumes them. Everything is registered to the bus.
 func P2PDependenciesPreRunE(cmd *cobra.Command, _ []string) error {
+
 	// TECHDEBT: this is to keep backwards compatibility with localnet
 	flags.ConfigPath = runtime.GetEnv("CONFIG_PATH", "build/config/config.validator1.json")
+	configs.ParseConfig(flags.ConfigPath)
+
+	// set final `remote_cli_url` value; order of precedence: flag > env var > config > default
+	flags.RemoteCLIURL = viper.GetString("remote_cli_url")
```

Comment:
Wow, literally have the same change in #902

Reply:
I liked your solution better than what I had done. :sweat_smile: I was trying to consolidate usages of `ParseConfig()` accross the CLI but later discovered that we're not constructing or referencing the config in the same way everywhere.

I think there's probably an issue that needs to be opened about:

1. Clearer separation between CLI and API code
    - In my view, `configs.ParseConfig()` is API code, it feels wrong to me to have all the viper calls that we currently do in there; viper shouldn't be part of our API dependency graph.
2. Parsing the config consistently in all CLI commands
  - Here, we're using the runtime manager to parse files at a dynamic path but [elsewhere](https://github.com/pokt-network/pocket/blob/main/app/client/cli/cmd.go#L50) we're assigning (and [referencing](https://github.com/pokt-network/pocket/blob/main/app/client/cli/utils.go#L314)) a global config var instead, relying on the unconventional behavior in 1.
3. Handling flags consistently in all commands
  I feel like we should have one place where we do `flags.RemoteCLIURL = ...` (and probably all the rest of the viper calls too) rather than duplicating it in multiple pre-run functions. I think we also need to ensure we're consistently overriding flags with viper correctly.

Reply:
#907 907